angular/core/components/invitation_form/invitation_form.coffee

```diff
@@ -21,17 +24,30 @@ angular.module('loomioApp').factory 'InvitationForm', ->
       $scope.addCustomMessageClicked = true
 
     $scope.invitees = ->
-      _.compact $scope.form.emails.split(' ')
+      $scope.form.emails.match(/[^\s<,]+?@[^>,\s]+/g) or []
 
     $scope.maxInvitations = ->
       $scope.invitees().length > 100
 
-    $scope.getTeamLink = ->
-      ModalService.open TeamLinkModal, group: -> $scope.form.group()
+    $scope.submitText = ->
+      if $scope.form.emails.length > 0
+        'invitation_form.send'
+      else
+        'invitation_form.done'
+
+    $scope.invalidEmail = ->
+      $scope.invitees().length == 0 and $scope.form.emails.length > 0
 
     $scope.canSubmit = ->
       $scope.invitees().length > 0 and $scope.form.group()
 
+    $scope.copied = ->
+      FlashService.success('common.copied')
+
+    $scope.invitationLink = ->
+      return unless $scope.form.group().shareableInvitation()
```

Comment:
I think we have to make sure that $scope.form.group() is present here as well... or otherwise at least check out what the edge case of 'I'm logged in but can't actually invite someone to any of my groups' looks like